go/cmd/dolt/commands/sql.go

```diff
@@ -580,64 +604,126 @@ func processBatchQuery(ctx context.Context, query string, se *sqlEngine) error {
 		return fmt.Errorf("Error parsing SQL: %v.", err.Error())
 	}
 
-	switch sqlStatement.(type) {
-	case *sqlparser.Insert:
+	if canProcessAsBatchInsert(sqlStatement) {
 		_, rowIter, err := se.query(ctx, query)
 		if err != nil {
 			return fmt.Errorf("Error inserting rows: %v", err.Error())
 		}
-		defer rowIter.Close()
 
-		err = mergeInsertResultIntoStats(rowIter, &batchEditStats)
-		if err != nil {
-			return fmt.Errorf("Error inserting rows: %v", err.Error())
+		if rowIter != nil {
+			defer rowIter.Close()
+			err = mergeResultIntoStats(sqlStatement, rowIter, batchEditStats)
+			if err != nil {
+				return fmt.Errorf("Error inserting rows: %v", err.Error())
+			}
 		}
 
-		if batchEditStats.numRowsInserted%maxBatchSize == 0 {
+		if batchEditStats.numUpdates()%maxBatchSize == 0 {
```

Comment:
this seems odd to me.  This is only going to flush if it's an exact multiple of maxBatchSize?  shouldn't this be >= instead of % ?

Reply:
Fixed.